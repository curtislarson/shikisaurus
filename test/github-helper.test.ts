import { GithubPermalinkParser } from "../api/mod.ts";
import { assert, assertEquals, assertExists, assertSnapshot } from "./test-deps.ts";

Deno.test("GithubPermalinkParser", async (t) => {
  const helper = await GithubPermalinkParser.init();

  await t.step("parseHash single line number", () => {
    const parsed = helper.parseLineNumbersHash(new URL("http://foo.local/#L1"));
    assertExists(parsed);
    assertEquals(parsed.start, 1);
    assertEquals(parsed.end, undefined);
  });

  await t.step("parseHash line number range", () => {
    const parsed = helper.parseLineNumbersHash(new URL("http://foo.local/#L2-L3"));
    assertExists(parsed);
    assertEquals(parsed.start, 2);
    assertEquals(parsed.end, 3);
  });

  await t.step("processUrl public repo blob entire file", async () => {
    const { content } = await helper.processUrl(
      new URL(
        "https://github.com/shikijs/shiki/blob/154db14d584c61bc86d5163f217d4609b662b5d6/packages/shiki/src/highlighter.ts",
      ),
    );

    assert(content.split("\n").length > 200, "Is a decently sized file so likely correct");
  });

  await t.step("processUrl public repo blob with line numbers", async () => {
    const { content } = await helper.processUrl(
      new URL(
        "https://github.com/shikijs/shiki/blob/154db14d584c61bc86d5163f217d4609b662b5d6/packages/shiki/src/highlighter.ts#L6-L11",
      ),
    );

    await assertSnapshot(t, content);
  });
});
