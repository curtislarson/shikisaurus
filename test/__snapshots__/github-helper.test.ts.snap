export const snapshot = {};

snapshot[`GithubPermalinkParser 1`] = `
"  IShikiTheme,
  IThemeRegistration,
  StringLiteralUnion
} from './types'
import { Resolver } from './resolver'
import { tokenizeWithTheme } from './themedTokenizer'"
`;
